src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariantContext.java

```diff
@@ -0,0 +1,280 @@
+        }
+    }
+
+    private static void checkIsStructuralVariantContext(final VariantContext vc) {
+        Utils.nonNull(vc, "the input variant context must not be null");
+        Utils.nonNull(vc.getAttributeAsString(VCFConstants.SVTYPE, null), "the SVTYPE annotation is missing");
+    }
+
+    private StructuralVariantContext(final VariantContext other) {
+        super(other);
+    }
+
+    /**
+     * The end position of this variant context.
+     * This end position is determined with this priorityq:
+     * <ul>
+     *     <li>the value of the annotation {@value VCFConstants#END_KEY} if present, otherwise</li>
+     *     <li>the value that results from applying the length annotated under {@value  GATKSVVCFConstants#SVLEN} based on the structural type if such annoation is present, otherwise</li>
+     *     <li>the variant context start position + the length of the reference allele - 1.</li>
+     * </ul>
+     * @return whatever is returned by {@link #getStart()} or greater.
+     */
+    @Override
+    public int getEnd() {
+        if (end <= 0) {
+            if (hasAttribute(VCFConstants.END_KEY)) {
+                end = getAttributeAsInt(VCFConstants.END_KEY, -1);
+            } else if (getStructuralVariantType() == StructuralVariantType.DEL) {
+                end = getStart() + getStructuralVariantLength();
+            } else {
+                end = super.getEnd();
+            }
+        }
+        return end;
+    }
+
+    /**
+     * Returns the assembly ids for this context's structural variant.
+     * <p>
+     *     The list returned is an immutable list.
+     * </p>
+     * @throws IllegalStateException if the {@link GATKSVVCFConstants#CONTIG_NAMES} annotation contains
+     * undefined contig names (.)
+     *
+     * @return never {@code null}, an empty list if no structural variant is specified.
+     */
+    public List<String> getContigNames() {
+        if (!hasAttribute(GATKSVVCFConstants.CONTIG_NAMES)) {
+            return Collections.emptyList();
+        } else {
+            final List<String> contigNames = getAttributeAsStringList(GATKSVVCFConstants.CONTIG_NAMES, null);
+            if (contigNames.contains(null)) {
+                throw new IllegalStateException("the contig names annotation contains undefined values");
+            }
+            return contigNames;
+        }
+    }
+
+    /**
+     * Composes the haplotype that corresponds to an allele based on the reference sequence.
+     * <p>
+     *     The resulting haplotype will be annotated with the appropriate Cigar and genome location.
+     * </p>
+     * @param index the index of the target allele.
+     * @param paddingSize extra bases from the reference sequence to be added on either side.
+     * @param reference the reference to use as source.
+     * @return never {@code null}.
+     */
+    public Haplotype composeHaplotypeBasedOnReference(final int index, final int paddingSize, final ReferenceMultiSource reference)  {
```

Comment:
My bad. I just talked with the Engine team yesterday and they suggest, for tool developers, when `PipelineOptions` are needed, always use `null` and they'll have a smart way of dealing with the case when `null` is not enough. So please keep it the old way. Sorry!